### PR TITLE
chore: update Lerna to version 6

### LINKED ISSUE #23688: Upgrade to Lerna 5+ with Nx
### What would you like?

I would like to propose we upgrade the repo to use Lerna 5 and enable Nx. This will enable Nx's caching capabilities as well as bring it up to date with the most recent version of Lerna. 

https://blog.nrwl.io/lerna-used-to-walk-now-it-can-fly-eab7a0fe7700

### Why is this needed?

This will speed up our builds by leveraging Nx caching and update Lerna to be on the most recent version.

### Other

_No response_

### PR DESCRIPTION
**Pull request by jordanpowell88**
- Closes #23688 

### Additional details
This updates Lerna from version 3 to version 6

### Steps to test
Any package.json script using lerna should still function as expected.

### How has the user experience changed?
n/a

### PR Tasks
- [ ] Have tests been added/updated?
- [ ] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [ ] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?

### CODE REVIEW

File: lerna.json [hunk 1]
```diff
@@ -8,5 +8,6 @@
     "system-tests"
   ],
   "useWorkspaces": true,
+  "useNx": false,
```
Comment (jordanpowell88) [inline lerna.json]: Once we get to a point where we can flip this switch we will start to see some significant DX and CI improvements! 😄

File: package.json [hunk 2]
```diff
@@ -207,7 +207,7 @@
     "through2": "^4.0.2",
     "tree-kill": "1.2.2",
     "ts-node": "^10.9.1",
-    "typescript": "^4.7.4",
+    "typescript": "4.7.4",
```
Comment (emilyrohrbough) [inline package.json:210]: why was this needed?

File: npm/create-cypress-tests/package.json [hunk 3]
```diff
@@ -25,14 +25,14 @@
     "find-up": "5.0.0",
     "fs-extra": "^9.1.0",
     "glob": "^7.1.6",
-    "inquirer": "7.3.3",
+    "inquirer": "8.2.4",
     "ora": "^5.1.0",
```
Comment (emilyrohrbough) [inline npm/create-cypress-tests/package.json:29]: why was this needed?

Comment (jordanpowell88) [reply npm/create-cypress-tests/package.json:29]: This was necessary as the Lerna upgrade has a dependency upon `inquirer`. 

See this [release](https://github.com/lerna/lerna/releases/tag/v5.2.0)

It was throwing errors when building by leaving it at `7.3.3`

File: package.json [hunk 4]
```diff
@@ -174,7 +174,7 @@
     "inquirer": "3.3.0",
     "inquirer-confirm": "2.0.3",
     "lazy-ass": "1.6.0",
-    "lerna": "3.22.1",
+    "lerna": "6.2.0",
```
Comment (emilyrohrbough) [inline package.json]: Can we use an open range?

File: npm/create-cypress-tests/package.json (hunk 3 above)
Comment (emilyrohrbough) [reply npm/create-cypress-tests/package.json:29]: What were the inquire breaking changes? Where the scripts in our repo using this dependency tested/check to ensure no other changes were needed?

Comment (jordanpowell88) [reply npm/create-cypress-tests/package.json:29]: [Version 8 dropped support for Node 10](https://github.com/SBoudrias/Inquirer.js/releases/tag/inquirer%408.0.0)

**Comment by astone123** [comment]
closing in favor of https://github.com/cypress-io/cypress/pull/26660
